eclair-core/src/main/scala/fr/acinq/eclair/io/ReconnectionTask.scala

```diff
@@ -0,0 +1,189 @@
+
+    case Event(FSM.Transition(_, Peer.DISCONNECTED, Peer.CONNECTED), _) =>
+      log.info("peer is connected")
+      goto(IDLE) using Nothing
+  }
+
+  when(WAITING) {
+    case Event(TickReconnect, d: WaitingData) =>
+      // we query the db every time because it may have been updated in the meantime (e.g. with network announcements)
+      getPeerAddressFromDb(nodeParams.db.peers, nodeParams.db.network, remoteNodeId) match {
+        case Some(address) =>
+          val connection = connect(address, origin_opt = None)
+          goto(CONNECTING) using ConnectingData(connection, address, d.nextReconnectionDelay)
+        case None =>
+          // we don't have an address for that peer, nothing to do
+          goto(IDLE) using Nothing
+      }
+
+    case Event(FSM.Transition(_, Peer.DISCONNECTED, Peer.CONNECTED), _) =>
+      log.info("peer is connected")
+      cancelTimer(RECONNECT_TIMER)
+      goto(IDLE) using Nothing
+  }
+
+  when(IDLE) {
+    case Event(FSM.Transition(_, previousState@(Peer.INSTANTIATING | Peer.CONNECTED), Peer.DISCONNECTED), _) =>
+      if (nodeParams.autoReconnect) {
+        // The random randomizeDelay does two things:
+        // - it adds a minimum delay. This delay is important in the case of a first connection to a new peer which advertises
+        // a public address. Right after the peer and this reconnection actor will be created, there will be a race between
+        // this automated connection task that uses data from network db, and the Peer.Connect command that may use the same
+        // address or a different one. This delay will cause the automated connection task to lose the race and prevent
+        // unnecessary parallel connections
+        // - it also adds some randomization. This is important in the case of a reconnection. If both peers have a public
+        // address, they may attempt to simultaneously connect back to each other, which could result in reconnection loop,
+        // given that each new connection will cause the previous one to be killed.
+        val initialDelay = randomizeDelay(nodeParams.initialRandomReconnectDelay)
+        val firstNextReconnectionDelay = previousState match {
+          case Peer.INSTANTIATING =>
+            // When restarting, we will ~immediately reconnect, but then:
+            // - we don't want all the subsequent reconnection attempts to be synchronized (herd effect)
+            // - we don't want to go through the exponential backoff delay, because we were offline, not them, so there is no
+            // reason to eagerly retry
+            // That's why we set the next reconnection delay to a random value between MAX_RECONNECT_INTERVAL/2 and MAX_RECONNECT_INTERVAL.
+            nodeParams.maxReconnectInterval.minus(Random.nextInt(nodeParams.maxReconnectInterval.toSeconds.toInt / 2).seconds)
+          case Peer.CONNECTED =>
+            log.info("peer is disconnected")
```

Comment:
It feels like this log is out of place, maybe put it in the Peer's transition from CONNECTED -> DISCONNECTED? Same for the other `log.info` about transitions.
It feels like you used these log lines to verify the state transitions of the reconnectTask, right? In that case you could instead add a `log.debug` on all state transitions of this actor?

Reply:
It's there because I didn't want to print "peer is disconnected" on startup. Those logs should be debug anyway. An equivalent log already exists in `Peer`.